cf-key/cf-key-functions.c

```diff
@@ -120,10 +120,23 @@ void ParseKeyArg(char *keyarg, char **filename, char **ipaddr, char **username)
 
 extern bool cf_key_interrupted;
 
-bool ShowHost(const char *hostkey, const char *address, bool incoming,
-                     const KeyHostSeen *quality, void *ctx)
+#define HOST_FMT_TRUNCATE "%-10.10s %-40.40s %-25.25s %-26.26s %-s\n"
+#define HOST_FMT_NO_TRUNCATE "%s\t%s\t%s\t%s\t%s\n"
+
+typedef struct _HostPrintState
+{
+    int count;
+    bool truncate;
+} HostPrintState;
+
+static bool ShowHost(
+    const char *hostkey,
+    const char *address,
+    bool incoming,
+    const KeyHostSeen *quality,
+    void *ctx)
```

Comment:
This doesn't have to be `void *`, right?

Reply:
NACK. The callback is defined with a void pointer in the type, and used in other places.